### PR TITLE
Mime additions (Hiccup HTML, BufferedImage) and README examples

### PR DESCRIPTION
**Pull request by aria42**
This PR probably has a few too many things:
* Bump to version `0.2.0-SNAPSHOT` since there's a breaking change
* change to `core.clj` to dynamically resolve loading Cider middleware so that you can use `cider-jack-in` without the issue that arises (see [this issue](https://github.com/clojure-emacs/cider-nrepl/issues/447))
* Replace the `IncanterPlot` class and instead generically extend the mime protocol to Java `BufferedImage` which works for Incanter and basically any JVM charting library which can spit out an image
* `add-javascript` in `helper` namespace to make it easier to use external JS
* Add Hiccup in `display` namespace to allow for rendering HTML/SVG from just hiccup data structures. 
* Changes to the README to link to new example notebooks and has screen-shots (check out the rendered version [here](https://github.com/aria42/clojupyter/blob/bb0aecf77e3597c3ad66b52f7d6379240f240fc5/README.md))
* Tests for mime convertible changes

### CODE REVIEW

File: src/clojupyter/misc/display.clj [hunk 1]
```diff
@@ -1,70 +1,68 @@
...
+  mc/PMimeConvertible
+  (to-mime [_]
+    (mc/stream-to-string
+     {:text/html html})))
 
+(defn html [html-str]
+  (HtmlString. html-str))
 
-;; Html
+(defn ^:deprecated make-html
```
Comment (didiercrunch) [inline src/clojupyter/misc/display.clj]: I do not see why `make-html` is depreciated. Surly it is less practical than `hiccup-html` but still totally valid and useful.

Comment (aria42) [reply src/clojupyter/misc/display.clj]: Oh I added `html` as an alias effectively. The `make-*` seemed redundant so the behavior of `display/html` will work exactly as `display/make-html` will

Comment (aria42) [reply src/clojupyter/misc/display.clj]: I may actually make it so there's only `display/html` and it converts to hiccup on vectors and leaves as is for string. Might be a lot of magic, but I think we want a really simple API for people.

Comment (didiercrunch) [reply src/clojupyter/misc/display.clj]: I agree at making `display/html` magic.

**Pull request merged**